### DISCUSSION TITLE
Display prespecified subset of questions

### DISCUSSION THREAD

**Discussion by zhenduowen** [PrairieLearn/PrairieLearn#8655] [Q&A] [answered]
Hi,

I have noticed that we have a prompt "(Choose all questions)" next to the zone title in an assessment. I also know that we can choose a random subset of the question "alternatives". Is it possible to display/release a prespecified subset of the questions?

**Comment by nwalters512** [comment]
Could you share a little more about what you're trying to do? Including a concrete example would be especially helpful!

**Reply by zhenduowen** [reply]
We are trying to hide certain questions in an assessment. We have many questions and each year we release a subset of these questions to the students. I thought we could hide questions because I saw the prompt "(choose all questions)" in the zone title. But I searched through the documentation, I am still not sure what this prompt says. 

My solution for now is to keep the assessment unreleased, make a copy as the released assessment, but only include a subset of questions in the released assessment.

![zone_title](https://github.com/PrairieLearn/PrairieLearn/assets/69138298/5472c3df-e861-48a0-bd0f-055c8521afa2)

**Reply by nwalters512** [reply, marked as answer]
The "Choose all questions" label indicates that all questions from the zone will be shown to students. As you pointed out in your original post, you can also specify a number of alternatives and have PrairieLearn pick and/or grade a subset of them, in which case the label will be different. In case it helps, here is where that label and its other variations are generated:

https://github.com/PrairieLearn/PrairieLearn/blob/baa1e3bcaa949e1be340fd6557f0de87a1a99bea/apps/prairielearn/src/pages/instructorAssessmentQuestions/instructorAssessmentQuestions.ejs#L63-L75

There's no concept of hiding a question from an assessment. If you don't want a question to appear in an assessment, simply don't include it in the assessment. Your solution of a template assessment sounds reasonable.

**Reply by zhenduowen** [reply]
I see. Thanks for your clarification!
